server/mscalendar/settings_daily_summary.go

```diff
@@ -0,0 +1,245 @@
+	err := s.store.SetSetting(userID, s.id, value)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *dailySummarySetting) Get(userID string) (interface{}, error) {
+	value, err := s.store.GetSetting(userID, s.id)
+	if err != nil {
+		return "", err
+	}
+
+	dsum, ok := value.(*store.DailySummarySettings)
+	if !ok {
+		return "", errors.New("current value is not a Daily Summary Setting")
+	}
+
+	return dsum, nil
+}
+
+func (s *dailySummarySetting) GetID() string {
+	return s.id
+}
+
+func (s *dailySummarySetting) GetTitle() string {
+	return s.title
+}
+
+func (s *dailySummarySetting) GetDescription() string {
+	return s.description
+}
+
+func (s *dailySummarySetting) GetDependency() string {
+	return s.dependsOn
+}
+
+func (s *dailySummarySetting) GetSlackAttachments(userID, settingHandler string, disabled bool) (*model.SlackAttachment, error) {
+	title := fmt.Sprintf("Setting: %s", s.title)
+	currentValueMessage := "Disabled"
+
+	actions := []*model.PostAction{}
+	if !disabled {
+		dsumRaw, err := s.Get(userID)
+		if err != nil {
+			return nil, err
+		}
+
+		currentH := "0"
+		currentM := "00"
+		currentAPM := "AM"
+		fullTime := "0:00AM"
+		currentEnable := false
+		currentTextValue := "Not set."
+
+		if dsumRaw != nil {
+			dsum := dsumRaw.(*store.DailySummarySettings)
+			fullTime = dsum.PostTime
+			currentEnable = dsum.Enable
+			splitted := strings.Split(fullTime, ":")
+			currentH = splitted[0]
+			currentM = splitted[1][:2]
+			currentAPM = splitted[1][2:]
+			enableText := "Disabled"
+			if currentEnable {
+				enableText = "Enabled"
+			}
+			currentTextValue = fmt.Sprintf("%s (%s) (%s)", dsum.PostTime, dsum.Timezone, enableText)
+		}
+
+		currentValueMessage = fmt.Sprintf("Current value: %s", currentTextValue)
+
+		actionOptionsH := model.PostAction{
+			Name: "H:",
+			Integration: &model.PostActionIntegration{
+				URL: settingHandler,
+				Context: map[string]interface{}{
+					settingspanel.ContextIDKey: s.id,
+				},
+			},
+			Type:          "select",
+			Options:       s.makeHOptions(currentM, currentAPM),
+			DefaultOption: fullTime,
+		}
+
+		actionOptionsM := model.PostAction{
+			Name: "M:",
+			Integration: &model.PostActionIntegration{
+				URL: settingHandler,
+				Context: map[string]interface{}{
+					settingspanel.ContextIDKey: s.id,
+				},
+			},
+			Type:          "select",
+			Options:       s.makeMOptions(currentH, currentAPM),
+			DefaultOption: fullTime,
+		}
+
+		actionOptionsAPM := model.PostAction{
+			Name: "AM/PM:",
+			Integration: &model.PostActionIntegration{
+				URL: settingHandler,
+				Context: map[string]interface{}{
+					settingspanel.ContextIDKey: s.id,
+				},
+			},
+			Type:          "select",
+			Options:       s.makeAPMOptions(currentH, currentM),
+			DefaultOption: fullTime,
+		}
+
+		actions = []*model.PostAction{&actionOptionsH, &actionOptionsM, &actionOptionsAPM}
+
+		if currentTextValue != "Not set" {
```

Comment:
`currentTextValue` is initialized as `Not set.` but is being compared to `Not set` here.